### PR TITLE
update closure compiler to v20170409

### PR DESCRIPTION
**Pull request by erwinmombay**
I can't upgrade to latest since everything breaks with the latest version of CC but i need to upgrade it to a version that has the updated eventtarget externs

### CODE REVIEW

File: extensions/amp-analytics/0.1/analytics-root.js [hunk 1]
```diff
@@ -146,7 +146,7 @@ export class AnalyticsRoot {
    * has not been requested before, it will be created.
    *
    * @param {string} name
-   * @param {function(new:./events.EventTracker, !AnalyticsRoot)} klass
+   * @param {function(new:./events.CustomEventTracker, !AnalyticsRoot)|function(new:./events.ClickEventTracker, !AnalyticsRoot)|function(new:./events.SignalTracker, !AnalyticsRoot)|function(new:./events.IniLoadTracker, !AnalyticsRoot)|function(new:./events.VideoEventTracker, !AnalyticsRoot)|function(new:./events.VideoEventTracker, !AnalyticsRoot)|function(new:./events.VisibilityTracker, !AnalyticsRoot)} klass
```
Comment (erwinmombay) [inline extensions/amp-analytics/0.1/analytics-root.js]: @cramforce @dvoytenko any other way to express this.

the old parameter type was invalid as abstract classes (EventTracker) aren't `instantiable`

File: .travis.yml [hunk 2]
```diff
@@ -56,10 +55,10 @@ matrix:
         jwt:
           secure: "Wze0F0vGL0UcxryOx1n/vcuD5LIMGyR+69Nc6IWLoRvZBbbIpFwVFhDE6rE9ranIXiA2Hc684N4sV8ASfNDF8RRSB+jyLov159qwgji2rBxIfQ/4kuDV2vYoAJvYMz8m42kwx5FV2VV9awqMMt8mwU3wYIrKIaVCxB34uV86KIlDlbrHxt17Bm5EIiUmwi9r1AAnW/63vVRUN264D77oB4j9UQ759PfD6BDwEt54O87KurNIaLseNCr1IvzfL8veEsZ3uTbLC1GtgHfR4IGgkS2YyN2QIk06VZWeRDEOalS3RcY0nDkbCmBywxIGObnrpEMzOpjBiOb2fxLoLvvpjlla5W84zJGfWE6q4T9IvkyHuDJE+sft5B+arjMIeA6PIeUhKdV27+6qqDEf7fILZ/U/Ekn9ds4zSV8hekAZPUyyPncOeyWppCIJ8sOeCrsebkRjH1BoX/d+FE+nP0bN/XkBpIi/nManx5FyS/kqjQWGKmvsFQfEWlSUaZi7XtEQEjvBizRkzvpJanSDaoiTDS2Keulmwii3XRId51FuGtnfDZFeggLaMTKGfBX9DlPkccwYAZe6vPNfYk1pNgEj6AtnifEhYVEO+aAuWhEnJ86od+1wDOL/h+a2XY6h8/gFBywsD95p7sXPfdVDCKgwagiBo+Hw5MNjztVF7lszg1A="
 # Temporarily disabling caching due to a new image that broke us. See #11204
```
Comment (rsimha-amp) [inline .travis.yml]: If this works, remove this comment.

File: extensions/amp-analytics/0.1/analytics-root.js (hunk 1 above)
Comment (dvoytenko) [reply extensions/amp-analytics/0.1/analytics-root.js]: Hmm. Obviously this is very bad. Do you think we can try to make templates work here? E.g.

```
/**
 * @param {string} name
 * @param {function(new:T, ...)} klass
 * @return {T}
 * @template T
 */
getTracker(name, klass) {}
```
?

Would be even better if we could do a template constraint like `@template T extends EventTracker`, but I've never seen template constraints in closure.

Comment (dvoytenko) [reply extensions/amp-analytics/0.1/analytics-root.js]: Yeah, just found this in [closure docs](https://github.com/google/closure-compiler/wiki/Generic-Types). 

> We do not currently support bounded generics (e.g. Foo<T extends Bar>) though we may in the future.

Too bad, but template should still work. We can (a) document this constraints, and (b) `dev.assert(instance instanceof EventTracker)` once the instance is created.

File: .travis.yml (hunk 2 above)
Comment (rsimha-amp) [reply .travis.yml]: Looks like the new closer compiler wasn't able to get rid of the JVM errors we saw with the new Travis image?

Comment (erwinmombay) [reply .travis.yml]: unfortunately did not

**Comment by rsimha-amp** [comment]
This PR fixes #11229

**Pull request merged**